scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -20,7 +20,7 @@ trait GrantHandler {
    */
   def clientCredentialRequired = true
 
-  def handleRequest[U](request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
+  def handleRequest[U](validatedClientId: Option[String], request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
```

Comment:
I was torn on this whole thing - what I was trying to avoid was the TokenEndpoint already has to validate a client (https://github.com/nulab/scala-oauth2-provider/blob/master/scala-oauth2-core/src/main/scala/scalaoauth2/provider/TokenEndpoint.scala#L13) so when the handleRequest is called - we need some way of telling the request handler that the client is already been validated and this is what the clientId is. If not, that means most of the handlers will just have to ask to validate the client again, even though functionally it was just validated a few lines back in the TokenEndpoint. Suggestions? the naming of things was tripping me up here.